sdk-extensions/incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfigurationParseTest.java

```diff
@@ -634,6 +634,9 @@ private static java.util.stream.Stream<Arguments> envVarSubstitutionArgs() {
         // Multiple environment variables referenced
         Arguments.of("key1: ${STR_1}${STR_2}\n", mapOf(entry("key1", "value1value2"))),
         Arguments.of("key1: ${STR_1} ${STR_2}\n", mapOf(entry("key1", "value1 value2"))),
+        Arguments.of(
+            "key1: ${STR_1} ${NOT_SET:-default} ${STR_2}\n",
+            mapOf(entry("key1", "value1 default value2"))),
```

Comment:
Without this fix, this resolves to `value1 default} ${STR_2`.

The regex isn't able to differentiate between the `}` ending the second env var reference and the `}` ending the third reference.